### PR TITLE
feat: Upgrading to React 18

### LINKED ISSUE #80: Update to React 18
Update UI project to React 18 and try it, do a good proofs to ensure no affecting current stuffs. It looks as a good time to try it because of project is starting.

The idea is can take advantage of new features of this new version. Besides that, can enjoy of a better performance it provide us.

### PR DESCRIPTION
**Pull request by german-mergestat**
Upgrading project to React 18. (Resolve #80)

- Updating some types with `children` properties to specify `React.ReactNode` (Mandatory in React 18)
- Using new `useId()` hook to replace `Math.random()`
- Fixing some staffs in linting file so don't get an error when run `"npm run lint"`
- Fixing way of building `titles` because was throwing an error detected with this new React version.

### CODE REVIEW

File: ui/src/views/repositories/repositories-view.tsx [hunk 1]
```diff
@@ -103,3 +103,5 @@ export const RepositoriesView: React.FC = () => {
     </main>
   )
 }
+
+export default RepositoriesView
```
Comment (patrickdevivo) [inline ui/src/views/repositories/repositories-view.tsx:107]:
```suggestion
export default RepositoriesView

```

Comment (german-mergestat) [reply ui/src/views/repositories/repositories-view.tsx]: Done [880526a](https://github.com/mergestat/fuse/pull/123/commits/880526a7bf3ff8895c9339f56c36ba3af6ec77ca)

File: ui/package-lock.json [hunk 2]
```diff
@@ -1,9609 +1,8 @@
 {
   "name": "fuse",
   "version": "0.1.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
Comment (patrickdevivo) [inline ui/package-lock.json:4]: Just checking - did you add dependencies here with the version of npm in the `.nvmrc` file? I don't think it should should change the `lockfileVersion` unless an older version of npm was used

**Review by patrickdevivo — DISMISSED** [review]
looks really great, thanks for doing this!

Good to merge other than the `lockfileVersion` in the `package-lock.json`

**Review by patrickdevivo — APPROVED** [review]

**Pull request merged**